pkg/evaluator/evaluator_test.go

```diff
@@ -56,6 +56,35 @@ func TestParseDecl(t *testing.T) {
 	}
 }
 
+func TestFuncScope(t *testing.T) {
+	prog := `
+f := "🦊"
+
+func outer
+    f := "🐤"
+    print f
+    if true
+    	f := "🎈"
+	print f
```

Comment:
indentation seems a little off I think.

Reply:
ok. fixed I think. I had tab indenting when it was at 8-spaces, which perhaps you saw as 4 spaces depending on your github settings.